cmd/ceph-cosi-driver/cmd.go

```diff
@@ -33,57 +30,25 @@ import (
 const provisionerName = "ceph.objectstorage.k8s.io"
 
 var (
-	driverAddress = "unix:///var/lib/cosi/cosi.sock"
-	AccessKey     = ""
-	SecretKey     = ""
-	Endpoint      = ""
+	driverAddress = flag.String("driver-address", "unix:///var/lib/cosi/cosi.sock", "driver address for socket")
+	driverPrefix  = flag.String("driver-prefix", "", "prefix for cosi driver, e.g. <prefix>.ceph.objectstorage.k8s.io")
 )
 
-var cmd = &cobra.Command{
-	Use:           "ceph-cosi-driver",
-	Short:         "Kubernetes COSI driver for Ceph RGW",
-	SilenceErrors: true,
-	SilenceUsage:  true,
-	RunE: func(cmd *cobra.Command, args []string) error {
-		return run(cmd.Context(), args)
-	},
-	DisableFlagsInUseLine: true,
-}
-
 func init() {
-	viper.AutomaticEnv()
-	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
-
-	flag.Set("alsologtostderr", "true")
-	kflags := flag.NewFlagSet("klog", flag.ExitOnError)
-	klog.InitFlags(kflags)
-
-	persistentFlags := cmd.PersistentFlags()
-	persistentFlags.AddGoFlagSet(kflags)
-
-	stringFlag := persistentFlags.StringVarP
-
-	stringFlag(&driverAddress,
-		"driver-addr",
-		"d",
-		driverAddress,
-		"path to unix domain socket where driver should listen")
-
-	viper.BindPFlags(cmd.PersistentFlags())
-	cmd.PersistentFlags().VisitAll(func(f *pflag.Flag) {
-		if viper.IsSet(f.Name) && viper.GetString(f.Name) != "" {
-			cmd.PersistentFlags().Set(f.Name, viper.GetString(f.Name))
-		}
-	})
+	klog.InitFlags(nil)
 }
 
-func run(ctx context.Context, args []string) error {
-	identityServer, bucketProvisioner, err := driver.NewDriver(ctx, provisionerName)
+func run(ctx context.Context /*, args []string*/) error {
+	if *driverPrefix == "" {
+		*driverPrefix = os.Getenv("POD_NAMESPACE")
+	}
```

Comment:
I think it might be easier to maintain the driver and Rook integration with it longer-term if we get rid of POD_NAMESPACE checking entirely and make `--driver-prefix` a required flag. 

I'd prefer to force the small number of current users to change their configuration now, rather than maintaining complex backwards compatibility code forever.

Reply:
Made requested changes